pkg/engine/operator/operator.go

```diff
@@ -46,10 +46,10 @@ func GetOperatorFromStringPattern(pattern string) Operator {
 	if pattern[:len(NotEqual)] == string(NotEqual) {
 		return NotEqual
 	}
-	if match, _ := regexp.Match(`^(\d+(\.\d+)?)([^-]*)!-(\d+(\.\d+)?)([^-]*)$`, []byte(pattern)); match {
+	if match, _ := regexp.Match(`^(-\d|\d+(\.\d+)?)([^-]*)!-((-|\+)\d|\d+(\.\d+)?)([^-]*)$`, []byte(pattern)); match {
```

Comment:
Test with `+1-+5` too

Reply:
Please pay attention to capture groups, it would be nice to have just two capture groups.
It will make things easier if we want to return range boundaries.

Reply:
right

Reply:
> Please pay attention to capture groups, it would be nice to have just two capture groups. It will make things easier if we want to return range boundaries.

Noted

Reply:
Give a look at https://regex101.com/r/3oGHSX/1 maybe it can give ideas.

Reply:
@eddycharly
I changed the regex to this, to remove 2 more capture groups, wdyt:
 ^(-|\+?\d+(\.\d+)?)([^-]*)-(-|\+?\d+(\.\d+)?)([^-]*)$

Reply:
Still not 2 capture groups right ?

Reply:
@eddycharly i'm going go through simplifying it further

Reply:
@eddycharly I found the link you shared very useful and i'm kindof leaning towards with a just a little change
^([-\+]?\d+(?:\.\d+)?[A-Za-z]*)-([-\+]?\d+(?:\.\d+)?[A-Za-z]*)$

Reply:
You need 0 or more [A-Za-z]

Reply:
huh, copied the wrong one 

^([-\+]?\d+(?:\.\d+)?[A-Za-z]*)-([-\+]?\d+(?:\.\d+)?[A-Za-z]*)$

Reply:
Still not correct.
You probably want something like `[A-Za-z]*`

Reply:
Also `(?:.\d+)` should be `(?:\.\d+)` (you need to escape the `.`)

Reply:
Is `^([-|\+]?\d+(?:\.\d+)?[A-Za-z]*)-([-|\+]?\d+(?:\.\d+)?[A-Za-z]*)$` not correct ?

Reply:
@eddycharly it is
i had planned to change [-\|+] to [-\+] since the pipe symbol wouldn't appear there

Reply:
i just pushed the one above

Reply:
@eddycharly i think i haven't gotten the hang of kuttl tests since they are a bit new for me
I went through writing one for this change yesterday but couldn't get it to work